File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/comment/comment/direct/Java/Go/s378126458_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	sc := bufio.NewReader(os.Stdin)
	N, _ := strconv.Atoi(strings.TrimSpace(readLine(sc)))
	Q, _ := strconv.Atoi(strings.TrimSpace(readLine(sc)))
	s := readLine(sc)
	pre := make([]int, N+1)

	for i := 1; i < N; i++ {
		if s[i-1] == 'A' && s[i] == 'C' {
			pre[i] = 1
		}
	}

	for i := 1; i < N; i++ {
		pre[i] += pre[i-1]
	}

	for Q > 0 {
		l, _ := strconv.Atoi(strings.TrimSpace(readLine(sc)))
		r, _ := strconv.Atoi(strings.TrimSpace(readLine(sc)))
		r--
		fmt.Println(pre[r] - pre[l-1])
		Q--
	}
}

func readLine(sc *bufio.Reader) string {
	line, _ := sc.ReadString('\n')
	return strings.TrimRight(line, "\n")
}

// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/mad/Java/Python/s378126458_commented.py
def main():
    import sys
    input = sys.stdin.readline
    N, Q = map(int, input().split())
    s = input().strip()
    pre = [0] * (N + 1)
    
    for i in range(1, N):
        if s[i - 1] == 'A' and s[i] == 'C':
            pre[i] = 1
    
    for i in range(1, N):
        pre[i] += pre[i - 1]
    
    for _ in range(Q):
        l, r = map(int, input().split())
        r -= 1
        print(pre[r] - pre[l - 1])

if __name__ == "__main__":
    main()
# End of Code
